src/components/Pages/RequestPage/RequestPage.tsx

```diff
@@ -259,38 +237,24 @@ export const RequestPage = () => {
       }
 
       const signer = await provider.getSigner()
-      const result = await provider.send(requestRef.current?.method, requestRef.current?.params)
+      const result = await provider.send(requestRef.current?.method, requestRef.current?.params ?? [])
       getAnalytics()?.track(TrackingEvents.CLICK, {
         action: ClickEvents.APPROVE_WALLET_INTERACTION,
         method: requestRef.current?.method
       })
-      const fetchResult = await authServerFetch('outcome', {
-        requestId,
-        sender: await signer.getAddress(),
-        result
-      })
-
-      if (fetchResult.error) {
-        throw new Error(fetchResult.error)
-      } else {
-        setView(View.WALLET_INTERACTION_COMPLETE)
-      }
+      await authServerClient.current.sendSuccessfulOutcome(requestId, await signer.getAddress(), result)
+      setView(View.WALLET_INTERACTION_COMPLETE)
     } catch (e) {
       console.error('Wallet error', JSON.stringify(e))
       captureException(e, { tags: { isWeb2Wallet: isUserUsingWeb2Wallet } })
       const signer = await browserProvider.current?.getSigner()
       if (signer) {
         if (isRpcError(e)) {
-          await authServerFetch('outcome', {
-            requestId,
-            sender: await signer.getAddress(),
-            error: e.error
-          })
+          await authServerClient.current.sendFailedOutcome(requestId, await signer.getAddress(), e.error)
         } else {
```

Comment:
and then here something like 
```suggestion
        } if (signer && !isRpcError(e)) {
```

Reply:
Done!